tests/modules/selenium/test_selenium_builder.py

```diff
@@ -1218,3 +1218,169 @@ def test_conditions_missing_then(self):
+                        "actions": [
+                            {
+                                "loop": "i",
+                                "start": 1,
+                                "do": [
+                                    "clickById(dd)"
+                                ]
+                            }
+                        ]}]}}})
+
+        with self.assertRaises(TaurusConfigError) as context:
+            self.obj.prepare()
+
+        self.assertTrue('Loop must contain' in str(context.exception))
+
+    def test_loop_missing_start(self):
+        self.configure({
+            "execution": [{
+                "executor": "apiritif",
+                "scenario": "loc_sc"}],
+            "scenarios": {
+                "loc_sc": {
+                    "requests": [{
+                        "label": "la-la",
+                        "actions": [
+                            {
+                                "loop": "i",
+                                "end": 10,
+                                "do": [
+                                    "clickById(dd)"
+                                ]
+                            }
+                        ]}]}}})
+
+        with self.assertRaises(TaurusConfigError) as context:
+            self.obj.prepare()
+
+        self.assertTrue('Loop must contain' in str(context.exception))
+
+    def test_loop_missing_do(self):
+        self.configure({
+            "execution": [{
+                "executor": "apiritif",
+                "scenario": "loc_sc"}],
+            "scenarios": {
+                "loc_sc": {
+                    "requests": [{
+                        "label": "la-la",
+                        "actions": [
+                            {
+                                "loop": "i",
+                                "start": 1,
+                                "end": 10
+                            }
+                        ]}]}}})
+
+        with self.assertRaises(TaurusConfigError) as context:
+            self.obj.prepare()
+
+        self.assertTrue('Loop must contain' in str(context.exception))
+
+    def test_loop_step_defaults_to_1(self):
+        self.configure({
+            "execution": [{
+                "executor": "apiritif",
+                "scenario": "loc_sc"}],
+            "scenarios": {
+                "loc_sc": {
+                    "requests": [{
+                        "actions": [
+                            {
+                                "loop": "i",
+                                "start": 1,
+                                "end": 10,
+                                "do": [
+                                    "clickById(${i})"
+                                ]
+                            }
+                        ]}]}}})
+
+        self.obj.prepare()
+        with open(self.obj.script) as fds:
+            content = fds.read()
+
+        target_lines = [
+            "for i in [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]",
```

Comment:
I doubt explicit list is the best way to define values. Possibly we have to use range here

Reply:
Changed to range instead here...